dace/codegen/targets/rtl.py

```diff
@@ -0,0 +1,519 @@
+model->rst_i = 0;  // no reset
+model->clk_i = 0; // neg clock
+model->valid_i = 0; // not valid
+model->ready_i = 0; // not ready 
+model->eval();
+
+// init vector
+{vector_init}
+
+// reset design
+model->rst_i = 1;
+model->clk_i = 1; // rising
+model->eval();
+model->clk_i = 0; // falling
+model->eval();
+model->rst_i = 0;
+model->clk_i = 1; // rising
+model->eval();
+model->clk_i = 0; // falling
+model->eval();
+
+// simulate until in_handshakes = out_handshakes = num_elements
+bool read_input_hs = false, write_output_hs = false;
+int in_ptr = 0, out_ptr = 0;
+{num_elements}
+
+while (out_ptr < num_elements) {{
+
+    // increment time
+    main_time++;
+
+    // check if valid_i and ready_o have been asserted at the rising clock edge -> input read handshake
+    if (model->ready_o == 1 && model->valid_i == 1){{
+        read_input_hs = true;
+    }} 
+    // feed new element
+    if(model->valid_i == 0 && in_ptr < num_elements){{
+        std::cout << "feed new element" << std::endl;
+        {inputs}
+        model->valid_i = 1;
+    }}
+
+    // export element
+    if(model->valid_o == 1){{
+        std::cout << "export element" << std::endl;
+        {outputs}
+        model->ready_i = 1;
+    }}
+    // check if valid_o and ready_i have been asserted at the rising clock edge -> output write handshake
+    if (model->ready_i == 1 && model->valid_o == 1){{
+        write_output_hs = true;
+    }}
+
+    // positive clock edge
+    model->clk_i = !model->clk_i;
+    model->eval();
+
+    // report internal state
+    if(DEBUG){{
+        VL_PRINTF("[t=%lu] clk_i=%u rst_i=%u valid_i=%u ready_i=%u valid_o=%u ready_o=%u \\n", main_time, model->clk_i, model->rst_i, model->valid_i, model->ready_i, model->valid_o, model->ready_o);
+        VL_PRINTF("{internal_state_str}", {internal_state_var});
+        std::cout << std::endl;
+    }}
+
+    // check if valid_i and ready_o have been asserted at the rising clock edge
+    if (read_input_hs){{
+        // remove valid_i flag
+        std::cout << "remove read_input_hs flag" << std::endl;
+        model->valid_i = 0;
+        read_input_hs = false;
+    }}
+
+    // check if valid_o and ready_i have been asserted at the rising clock edge
+    if (write_output_hs){{
+        // remove valid_i flag
+        std::cout << "remove write_output_hs flag" << std::endl;
+        model->ready_i = 0;
+        write_output_hs = false;
+    }}
+
+
+    // negative clock edge
+    model->clk_i = !model->clk_i;
+    model->eval();
+}}
+
+// report internal state
+if(DEBUG){{
+    VL_PRINTF("[t=%lu] clk_i=%u rst_i=%u valid_i=%u ready_i=%u valid_o=%u ready_o=%u \\n", main_time, model->clk_i, model->rst_i, model->valid_i, model->ready_i, model->valid_o, model->ready_o);
+    VL_PRINTF("{internal_state_str}", {internal_state_var});
+    std::cout << std::endl;
+}}
+
+// final model cleanup
+model->final();
+
+// clean up resources
+delete model;
+model = NULL;
+
+std::cout << "SIM END" << std::endl;
```

Comment:
Same as above, be more descriptive (and perhaps make this dependent on verbosity flag)